bitcoin/src/address/error.rs

```diff
@@ -21,13 +21,6 @@ pub enum Error {
     ExcessiveScriptSize,
     /// Script is not a p2pkh, p2sh or witness program.
     UnrecognizedScript,
-    /// Address's network differs from required one.
-    NetworkValidation {
-        /// Network that was required.
-        required: Network,
-        /// The address itself
-        address: Address<NetworkUnchecked>,
-    },
```

Comment:
This was previously intentionally put here because the expected code is:

```rust
let address = input.parse::<Address<_>>()?.require_network(network)?;
```

Reply:
That code works fine without the `NetworkValidation` variant (for example, application code using `anyhow`). Or am I missing something?

Reply:
> application code using `anyhow`

Exactly. The idea is to not force a different error or even a different crate. While we try to separate errors lately, I believe in this case keeping it as-is is better. Unless we want a different `enum AddressError { Parse(ParseError), Network(NetworkValidationError) }` But I highly doubt it's usefulness.

Reply:
I'm not following you?

```rust
let address = input.parse::<Address<_>>()? // 1
    .require_network(network)?; // 2
```

1. Returns an error enum with a bunch of variants, one of which is never returned (validation)
2. Returns an error enum of which none of the variants are ever returned except the validation variant.

These seem like two totally different errors to me. What am I missing?

Reply:
I'm saying it's reasonable to have an exception from our general rule because it's expected people will just chain the calls and match on error in the caller. You won't have `parse` without `require_network` or vice-versa.

I'm assuming the function that snippet is written in returns `AddressError` or something that it can be converted into.